homeassistant/components/iaqualink/__init__.py

```diff
@@ -191,6 +210,14 @@ async def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+async def try_await(awaitable: Awaitable) -> None:
+    """Execute API call while catching service exceptions."""
+    try:
+        await awaitable
+    except AqualinkServiceException as svc_exception:
+        raise HomeAssistantError("Aqualink Exception raised") from svc_exception
```

Comment:
Can you give a better error here?

Reply:
I'm not sure what would be better in this case. The details of the error are wrapped in the AqualinkServiceException which is passed along.

Reply:
The user is likely only going to see the text in the UI so something like https://github.com/home-assistant/core/pull/53745/files#r772811878